src/vs/platform/extensionManagement/common/abstractExtensionManagementService.ts

```diff
@@ -240,22 +244,43 @@ export abstract class AbstractExtensionManagementService extends Disposable impl
 				}));
 			}
 
+			// Await for extension dependencies that were requested to install beforehand
+			// https://github.com/microsoft/vscode/issues/149309
+			const localDeps = await this.joinAllSettled(alreadyRequestedInstallExtensionTasks.map(task => task.waitUntilTaskIsFinished()));
+			alreadyRequestedInstallExtensionTasks.forEach((task, i) => installResults.push({ local: localDeps[i], identifier: task.identifier, operation: task.operation, source: task.source }));
```

Comment:
It is good that we are waiting until dependencies are installed which fixes one part of the issue but triggering the install event for the same extension multiple times might have side affects. I think the right solution here is to wait until the dependencies are completely installed ie., wait until the install events are triggered for them. I am thinking to come up with such solution.